### ISSUE TITLE
Env vars missing - root cause of 101 issue

### ISSUE THREAD

**Issue by NewtonChutney** [danielbeach/RustOnApacheAirflow#1]
Hello @danielbeach,
I think there are separate temporary environments for each task run.. I've found with bash operator, that each run is in a different temporary folder.. Maybe python operator is similar? 

UPDATE: 
I noticed you were using a particular temporary directory to store, so this shouldn't be an issue

And when you run on a set of Kubernetes workers, shouldn't the download, chmod, and execution be in the same task on the DAG, so they get executed inside the same worker?

**Comment by NewtonChutney** [comment]
This is after an initial analysis from your blog, will dive deeper and update you in 12hrs...
Currently travelling.. 🫠

**Comment by NewtonChutney** [comment]
![image](https://github.com/danielbeach/RustOnApacheAirflow/assets/70827815/42a65687-1404-4731-b85a-6931815ce0a1)
The issue is caused by the lack of env variables

Had you tried check_output.. it would have been obvious.. 😅
![image](https://github.com/danielbeach/RustOnApacheAirflow/assets/70827815/c37e2015-9a9c-4562-b132-e12d5a2be574)

**Comment by veloxl** [comment]
I was also going to suggest using `subprocess.check_output` rather than `subprocess.check_call` here:
```
subprocess.check_call(f"{temp_dir} {uri}", shell=True)
```
_More info on [stackoverflow](https://stackoverflow.com/questions/36169571/python-subprocess-check-call-vs-check-output)._

The TL;DR is `check_call()` starts the external process and returns immediately without waiting for it to exit. `check_output()` waits for the background python processes to complete and then exits. As @NewtonChutney mentioned, `check_output()` should also provide the output from the child processes which should make debugging easier.

Also, you might be able to pass env variables to the subprocess command like this:
```
subprocess.check_output(f"{temp_dir} {uri}", shell=True, env={"KEY": "VALUE"})
```

I haven't tested it but might be worth a look.

**Comment by NewtonChutney** [comment]
Yep, we have a similar setup, to inject env variables for bashOperator..
Currently, we're using it to run pytests inside Airflow's worker environment..
